src/RefractDataStructure.cc

```diff
@@ -456,24 +456,36 @@ namespace
     struct ExtractTypeDefinition {
 
         ElementData<T>& data;
-        const ConversionContext& context;
+        ConversionContext& context;
+
+        using TypeDefinition = NodeInfo<mson::TypeDefinition>;
 
         template <typename E, bool IsPrimitive = is_primitive<E>()>
         struct Fetch;
 
         template <typename E>
         struct Fetch<E, ComplexType> {
-            ElementInfo<T> operator()(const mson::TypeNames& typeNames, const ConversionContext& context) const
+            ElementInfo<T> operator()(const TypeDefinition& typeDefinition, ConversionContext& context) const
             {
                 std::deque<std::unique_ptr<IElement> > types;
 
+                auto typeNames = typeDefinition.node->typeSpecification.nestedTypes;
+
                 for (const auto& type : typeNames) {
                     mson::BaseTypeName typeName = type.base;
                     FactoryCreateMethod method = eValue;
 
                     if (typeName == mson::UndefinedTypeName && !type.symbol.literal.empty()) {
                         typeName = GetMsonTypeFromName(type.symbol.literal, context);
                         method = type.symbol.variable ? eSample : eElement;
+
+                        if (typeName == mson::UndefinedTypeName
+                            && method == eElement) { // undefined named element in hinting
+                            std::stringstream msg;
+                            msg << "Undefined named type '" << type.symbol.literal << "' referenced in type definition";
+                            context.warn(snowcrash::Warning(
```

Comment:
Should this be an error instead? We're producing an invalid parse result otherwise because there will still be the reference to the symbol, from `test/fixtures/render/issue-556.json`:

```json
{
  "element": "Unknown"
}
```

Reply:
Not sure if error, because type hinting is just info, content in array is  not mandatory. 
We can avoid creating element for referenced element

Reply:
We treat the following as an error also:

```apib
# API
## GET /

+ Response 200 (application/json)
    + Attributes
        + property (Unknown)
```

```
error: (4)  base type 'Unknown' is not defined in the document; line 6, column 11 - line 6, column 29
```

Reply:
We can continue in parsing, so I vote for warning

Reply:
Yes, can we make it so that `{ element: "Unknown" }` goes away from the parse result in the case of warning?

Reply:
simple change, just `continue` in condition. We can discuss it in stand up. 
If "err && stop" or "warn && remove referenced named element"

Reply:
Fixed

Reply:
Remove undefined from result && produce warning